x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/ensemble/WeightedMode.java

```diff
@@ -133,7 +146,7 @@ public String getName() {
 
     @Override
     public boolean compatibleWith(TargetType targetType) {
-        return true;
+        return targetType.equals(TargetType.CLASSIFICATION);
```

Comment:
Probably it's ok here but I'd reverse the order so that the constant is first (useful if we suspect null argument):
```
return TargetType.CLASSIFICATION.equals(targetType);
```